background.ts

```diff
@@ -826,6 +846,83 @@ ipcMainProxy.handle('show-message-box-rd', async(_event, options: Electron.Messa
   return response;
 });
 
+ipcMainProxy.handle('show-snapshots-dialog', async(
+  event,
+  options: { window: Partial<Electron.MessageBoxOptions>, format: SnapshotDialog },
+) => {
+  const mainWindow = window.getWindow('main');
+
+  const dialog = window.openDialog(
+    'SnapshotsDialog',
+    {
+      title:    options.format.type === 'question' ? 'Snapshots' : undefined,
+      modal:    true,
+      parent:   mainWindow || undefined,
+      frame:    options.format.type === 'question',
+      height:   options.format.type === 'question' ? 365 : 500,
+      center:   true,
+      closable: options.format.type === 'question',
+      movable:  options.format.type === 'question',
+    });
+
+  const onMainWindowMove = () => {
+    /** Lock dialog position */
+    if (mainWindow && dialog) {
+      const mainWindowPos = mainWindow.getPosition();
+
+      const mainWindowDim = mainWindow.getSize();
+      const dialogDim = dialog.getSize();
+
+      const x = Math.floor(mainWindowPos[0] + ((mainWindowDim[0] - dialogDim[0]) / 2));
+      const y = Math.floor(mainWindowPos[1]);
+
+      dialog.setPosition(x, y);
+    }
+  };
+
+  if (mainWindow && options.format.type !== 'question') {
+    mainWindow.on('move', onMainWindowMove);
+
+    if (options.format.type !== 'question') {
+      /** On Linux it does nothing */
+      mainWindow.setOpacity(0.7);
```

Comment:
I'm not sure if changing the opacity generates the desired effect here. The entire main window becomes transparent, where I'm thinking that the intent is to dim the main window as a means to communicate that it is "disabled" while the modal is open

![image](https://github.com/rancher-sandbox/rancher-desktop/assets/835961/c1734f90-1367-4c93-9002-e6c60c9e80ee)

We might need to create some form of overlay using an electron BrowserView[^1], via css, or some other method

[^1]: https://www.electronjs.org/docs/latest/api/browser-view

Reply:
Thank you Phil.

- I've fixed the close button on Windows; it was caused by `closable: true` option, which is not required in the snapshots case.
- I've fixed the dialog position on windows, now it should be always in the middle of the `mainWindow`.